### PR TITLE
SDK-1231: Attribute Issuance Details

### PR DESCRIPTION
**Pull request by MrBurtyyy**
## Requires
- [X] Merge 3.7.3 Hotfix into development

## Added
* `ActivityDetails`
  * `getExtraData()` - Returns ExtraData associated with the receipt
* `ExtraData`
  * `getAttributeIssuanceDetails()` - Returns the attribute issuance details if available from extra data
* `AttributeIssuanceDetails`
  * `getToken()`
  * `getExpiryDate()`
  * `getIssuingAttributes()`

### CODE REVIEW

File: tests/yoti_common/third.party.attribute.converter.spec.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        .convertThirdPartyAttribute(thirdPartyProto);
+
+      expect(thirdPartyAttribute).not.toBe(undefined);
+      expect(thirdPartyAttribute).toBeInstanceOf(AttributeIssuanceDetails);
+      expect(thirdPartyAttribute.getExpiryDate()).toBe(undefined);
+      expect(thirdPartyAttribute.getIssuingAttributes().length).toEqual(1);
+      expect(thirdPartyAttribute.getIssuingAttributes()[0]).toEqual('com.thirdparty.id');
+    });
+
+    it('should parse multiple attribute definitions', () => {
```
Comment (echarrod) [inline tests/yoti_common/third.party.attribute.converter.spec.js]: Doesn't look like this test is finished - seems to be the same as the first lines of the test above?

Comment (MrBurtyyy) [reply tests/yoti_common/third.party.attribute.converter.spec.js]: Fixed 👍

File: src/profile_service/activity.details.js [hunk 2]
```diff
@@ -35,11 +35,12 @@ class ActivityDetails {
    * @param {array} decryptedApplicationProfile
    *   Decrypted application profile data.
    */
-  constructor(parsedResponse, decryptedProfile, decryptedApplicationProfile) {
+  constructor(parsedResponse, decryptedProfile, decryptedApplicationProfile, extraData) {
```
[the remark this answers is not in the document]
Comment (davidgrayston) [reply src/profile_service/activity.details.js]: Added in 834f6f4

File: src/profile_service/extra.data.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return filtered[0];
+  }
+
+  return undefined;
+}
+
+class ExtraData {
+  /**
+   *
+   * @param {Object} attributeIssuanceDetails
```
[the remark this answers is not in the document]
Comment (davidgrayston) [reply src/profile_service/extra.data.js]: Fixed in 834f6f4

File: src/data_type/attribute.issuance.details.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const Validation = require('../yoti_common/validation');
+
+class AttributeIssuanceDetails {
+  constructor(token, expiryDate, issuingAttributes) {
```
[the remark this answers is not in the document]
Comment (davidgrayston) [reply src/data_type/attribute.issuance.details.js]: Added in 834f6f4

**Comment by davidgrayston** [comment]
Requesting review for changes 47c4d6f...834f6f4
- Decrypting ExtraData (re-organised `src/yoti_common/index.js` to accommodate this)
- Added JSDocs
- Added additional validation for constructor arguments
- `AttributeIssuanceDetails.getToken()` now returns base64 encoded token

**Pull request merged**